ckanext/stadtzh_losdharvest/harvester.py

```diff
@@ -65,225 +60,21 @@ def _is_published(self, date_str):
-        if not status == "delete":
-            if harvest_object.content is None:
-                self._save_object_error(
-                    "Empty content for object {0}".format(harvest_object.id),
-                    harvest_object,
-                    "Import",
-                )
-                return False
-
-            try:
-                dataset = json.loads(harvest_object.content)
-            except ValueError:
-                self._save_object_error(
-                    "Could not parse content for object {0}".format(harvest_object.id),
-                    harvest_object,
-                    "Import",
-                )
-                return False
-
-            # set harvest_object's status to 'delete' if the
-            # package's issue-date is in the future
-            if not self._is_published(dataset.get("dateFirstPublished", None)):
-                log.info(
-                    "Deleting dataset {} because it has not yet been published, "
-                    "according to the DCT.issued value at the source"
-                )
-                harvest_object.extras = [
-                    HarvestObjectExtra(key="status", value="delete")
-                ]
-                status = "delete"
-
-        if status == "delete":
-            # Delete package
-            context = {
-                "model": model,
-                "session": model.Session,
-                "user": self._get_user_name(),
-                "ignore_auth": True,
-            }
-
-            p.toolkit.get_action("package_delete")(context, {"id": harvest_object.guid})
-            log.info("Deleted package with guid {0}".format(harvest_object.guid))
-            return True
-
-        # Get the last harvested object (if any)
-        previous_object = (
-            model.Session.query(HarvestObject)
-            .filter(HarvestObject.guid == harvest_object.guid)
-            .filter(HarvestObject.current is True)
-            .first()
-        )
-
-        # Flag previous object as not current anymore
-        if previous_object:
-            previous_object.current = False
-            previous_object.add()
-
-        # Flag this object as the current one
-        harvest_object.current = True
-        harvest_object.add()
-
-        context = {
-            "user": self._get_user_name(),
-            "return_id_only": True,
-            "ignore_auth": True,
-        }
-
-        dataset = self.modify_package_dict(dataset, {}, harvest_object)
+    def after_parsing(self, rdf_parser, harvest_job):
+        """Called just after the content from the remote RDF file has been parsed
 
-        # Check if a dataset with the same guid exists
-        existing_dataset = self._get_existing_dataset(harvest_object.guid)
+        Filters the datasets in the parser to only include those that have been
+        published (according to the dateFirstPublished field).
+        """
+        all_datasets = rdf_parser.datasets()
 
-        try:
```

Comment:
I'm a little confused. Your explanation sounds good and makes sense to me. Happy that we can drop a lot of code, but can you explain to me, why the following block is not needed anymore?